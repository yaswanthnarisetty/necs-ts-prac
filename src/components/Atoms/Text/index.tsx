'use client'
import  React  from "react";
import styled from "@emotion/styled";


export type TextProps = {
  color?: string;
  fontSize?: string;
  fontWeight?: string;
  lineHeight?: string;
  margin?: string;
  padding?: string;
  mt?: string;
  mr?: string;
  mb?: string;
  ml?: string;
  paddingTop?: string;
  paddingRight?: string;
  paddingBottom?: string;
  paddingLeft?: string;
  fontFamily?: string;
};

const Text = styled.p<TextProps>((props) => ({
  color: props.color || "black",
  fontSize: props.fontSize || "10px",
  fontWeight: props.fontWeight || "20px",
  lineHeight: props.lineHeight || "2px",
  margin: props.margin || "0",
  padding: props.padding || "0",
  marginTop: props.mt || "0",
  marginRight: props.mr || "0",
  marginBottom: props.mb || "0",
  marginLeft: props.ml || "0",
  paddingTop: props.paddingTop || "0",
  paddingRight: props.paddingRight || "0",
  paddingBottom: props.paddingBottom || "0",
  paddingLeft: props.paddingLeft || "0",
  fontFamily: props.fontFamily || "",
}));

export default Text;
