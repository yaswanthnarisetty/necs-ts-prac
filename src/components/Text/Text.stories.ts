import type { Meta, StoryObj } from '@storybook/react';
import { Theme } from "@emotion/react";
import Text from './Text';
// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
    title: 'Components/Text',
    component: Text,
    parameters: {
        // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
        layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
    tags: ['autodocs'],
    argTypes: {
        children: { control: 'text' },
        fontSize: { control: 'text' },
        variant: { control: 'select', options: ['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'subtitle1', 'subtitle2', 'body1', 'body2'] },
    },
    // More on argTypes: https://storybook.js.org/docs/api/argtypes
} satisfies Meta<typeof Text>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
    args: {
        children: 'Default Text',
        variant: 'body1',
        fontSize: '1rem',
    }
};
// Add more variants as needed
