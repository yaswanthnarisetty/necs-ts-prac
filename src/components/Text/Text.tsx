import React from "react";
import styled from "@emotion/styled";
import { Theme } from "@emotion/react";

export type TextProps = {
  children: React.ReactNode;
  color?: string;
  variant?: keyof Theme["typography"];
  fontSize?: string;
};

const Text = styled.p<TextProps>`
  ${({ theme, variant }) =>
    theme.typography[variant as keyof Theme["typography"]]};
  color: ${({ color }) => color};
  font-size: ${({ fontSize }) => fontSize};
`;

export default Text;
