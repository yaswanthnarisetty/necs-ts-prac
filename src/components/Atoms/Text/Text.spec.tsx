import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import  Text  from "./index";

describe("Text Component", () => {
  test("renders with default text content", () => {
    render(<Text>Hello, World!</Text>);
    expect(screen.getByText("Hello, World!")).toBeInTheDocument();
  });

  test("renders with custom text content", () => {
    const customText = "Custom Text Content";
    render(<Text>{customText}</Text>);
    expect(screen.getByText(customText)).toBeInTheDocument();
  });

  test("applies default styles if no props are provided", () => {
    render(<Text>Hello, World!</Text>);
    const textElement = screen.getByText("Hello, World!");
    expect(textElement).toHaveStyle("color: black");
    expect(textElement).toHaveStyle("font-size: 10px");
  });

  test("applies custom styles when props are provided", () => {
    render(
      <Text
        color="red"
        fontSize="20px"
        margin="10px"
        padding="5px"
        fontFamily="Arial, sans-serif"
      >
        Custom Styles Text
      </Text>
    );

    const textElement = screen.getByText("Custom Styles Text");

    
    expect(textElement).toHaveStyle("color: red");
    expect(textElement).toHaveStyle("font-size: 20px");
    expect(textElement).toHaveStyle("margin: 10px");
    expect(textElement).toHaveStyle("padding: 5px");
    expect(textElement).toHaveStyle("font-family: 'Arial, sans-serif'");
  });

  // test("handles click event if onClick prop is provided", () => {
  //   const onClickMock = jest.fn();
  //   render(<Text onClick={onClickMock}>Clickable Text</Text>);

  //   const textElement = screen.getByText("Clickable Text");
  //   fireEvent.click(textElement);

  //   expect(onClickMock).toHaveBeenCalled();
  // });
});
