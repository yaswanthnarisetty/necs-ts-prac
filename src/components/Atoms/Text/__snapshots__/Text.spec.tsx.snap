// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Text Component > renders with custom color 1`] = `
<p
  class="css-16ehan8"
  color="red"
>
  Custom Color Text
</p>
`;

exports[`Text Component > renders with custom font family 1`] = `
<p
  class="css-1yusjdk"
  font-family="Arial, sans-serif"
>
  Custom Font Family Text
</p>
`;

exports[`Text Component > renders with custom font size 1`] = `
<p
  class="css-14ggzw9"
  font-size="20px"
>
  Custom Font Size Text
</p>
`;

exports[`Text Component > renders with custom margin and padding 1`] = `
<p
  class="css-9pjzk3"
>
  Custom Margin and Padding Text
</p>
`;

exports[`Text Component > renders with custom margin-top, margin-right, margin-bottom, and margin-left 1`] = `
<p
  class="css-83q1dl"
>
  Custom Margin Text
</p>
`;

exports[`Text Component > renders with custom padding-top, padding-right, padding-bottom, and padding-left 1`] = `
<p
  class="css-hglw8m"
>
  Custom Padding Text
</p>
`;

exports[`Text Component > renders with default styles if no props are provided 1`] = `
<p
  class="css-9ng7ju"
>
  Hello, World!
</p>
`;
